src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1740,6 +1742,32 @@ static bool processOnChangeConditionForUpdateContext
 
 /* ****************************************************************************
 *
+* setMetadataForFlags -
+*/
+static void setMetadataForFlags(ContextElementResponse* notifyCerP)
+{
+  for (unsigned int ix = 0; ix < notifyCerP->contextElement.contextAttributeVector.size(); ix++)
+  {
+    ContextAttribute* caP = notifyCerP->contextElement.contextAttributeVector[ix];
+
+    if (caP->onUpdate)
+    {
+      Metadata* newMdP = new Metadata(NGSI_MD_NOTIF_ONUPDATE, DEFAULT_ATTR_BOOL_TYPE, true);
+      caP->metadataVector.push_back(newMdP);
+    }
+
+    if (caP->onChange)
+    {
+      Metadata* newMdP = new Metadata(NGSI_MD_NOTIF_ONCHANGE, DEFAULT_ATTR_BOOL_TYPE, true);
+      caP->metadataVector.push_back(newMdP);
+    }
+  }
```

Comment:
> Yes, probably using "else if" is better thant "if" in line 1759. I'll change that.

Lokking closer, we cannot do an else, as both flags are independent. Thus, I think the current logic is correct or I don't understanding what do you mean by "adding an else may make the code easier to read and avoid a possible bug" :). In the latter case, a code sketch would help to understand.